code/test/VsTemplates.Test/VsTemplatesContentValidationTests.cs

```diff
@@ -0,0 +1,136 @@
+        private const string ProjectTemplatesRoot = "../../../../../src/ProjectTemplates";
+        private const string VsTemplateFileExtension = ".vstemplate";
+        private static readonly List<string> WizardProjectTemplatesRoot = new List<string>
+        {
+            "../../../../../src/ProjectTemplates/UWP",
+            "../../../../../src/ProjectTemplates/WPF",
+            "../../../../../src/ProjectTemplates/WinUI/Cpp/Cpp.WinUI.Desktop.Solution",
+            "../../../../../src/ProjectTemplates/WinUI/Cpp/Cpp.WinUI.UWP.Solution",
+            "../../../../../src/ProjectTemplates/WinUI/CS/CSharp.WinUI.Desktop.Solution",
+            "../../../../../src/ProjectTemplates/WinUI/CS/CSharp.WinUI.Uwp.Solution",
+        };
+
+        public static IEnumerable<object[]> GetWizardProjectTemplateFiles()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            foreach (var path in WizardProjectTemplatesRoot)
+            {
+                var filePaths = Directory.GetFiles(path, $"*{VsTemplateFileExtension}", SearchOption.AllDirectories);
+                vsTemplatesPaths.AddRange(filePaths.Select(p => new object[] { p }));
+            }
+
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsTemplatePaths()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            var vsItemTemplatesPaths = GetAllVsTemplatePaths(ItemTemplatesRoot).Select(p => new object[] { p });
+            var vsProjectTemplatesPaths = GetAllVsTemplatePaths(ProjectTemplatesRoot).Select(p => new object[] { p });
+            vsTemplatesPaths.AddRange(vsItemTemplatesPaths);
+            vsTemplatesPaths.AddRange(vsProjectTemplatesPaths);
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsTemplateFiles()
+        {
+            var vsTemplatesPaths = new List<object[]>();
+            var vsItemTemplatesPaths = GetAllVsItemTemplateFiles();
+            var vsProjectTemplatesPaths = GetAllVsProjectTemplateFiles();
+            vsTemplatesPaths.AddRange(vsItemTemplatesPaths);
+            vsTemplatesPaths.AddRange(vsProjectTemplatesPaths);
+            return vsTemplatesPaths;
+        }
+
+        public static IEnumerable<object[]> GetAllVsProjectTemplateFiles()
+        {
+            var filePaths = Directory.GetFiles(ProjectTemplatesRoot, $"*{VsTemplateFileExtension}", SearchOption.AllDirectories);
+            return filePaths.Select(p => new object[] { p });
+        }
+
+        private static IEnumerable<object[]> GetAllVsItemTemplateFiles()
+        {
+            var filePaths = Directory.GetFiles(ItemTemplatesRoot, $"*{VsTemplateFileExtension}", SearchOption.AllDirectories);
+            return filePaths.Select(p => new object[] { p });
+        }
+
+        private static IEnumerable<string> GetAllVsTemplatePaths(string directory)
```

Comment:
The name `GetAllVsTemplatePaths` is potentially confusing. I initially thought it was the pat to the actual files, but it's the path to the directory that contains them.
Having a name that clearly expresses what the method does and/or returns can save time and confusion when other people are looking at the code in the future.